api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -462,10 +469,17 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
 					log.warn("Found " + patients.size() + " patients with the same biometric id: " + subjectId);
 				}
 				for (Patient patient : patients) {
+					possessNationalFpId = possessNationalFpId(patient) || possessNationalFpId;
 					result.add(new PatientAndMatchQuality(patient, match.getMatchScore(),
-							Collections.singletonList("fingerprint")));
+							Collections.singletonList(LOCAL_FINGERPRINT)));
 				}
 			}
+
+			if (registrationCoreProperties.isMpiEnabled() && !possessNationalFpId) {
+				Patient mpiPatient = findMpiPatient(subjectId, nationalBiometricId.getUuid());
+				result.add(new PatientAndMatchQuality(mpiPatient,
+						match.getMatchScore(), Collections.singletonList(NATIONAL_FINGERPRINT)));
+			}
 		}
 
 		return result;
```

Comment:
findByBiometricMatch could be split into smaller methods with the descriptive method names. However no changes required, just a hint.